src/System.Linq/src/System/Linq/Contains.cs

```diff
@@ -9,27 +9,34 @@ namespace System.Linq
     public static partial class Enumerable
     {
         public static bool Contains<TSource>(this IEnumerable<TSource> source, TSource value) =>
-            source is ICollection<TSource> collection
-            ? collection.Contains(value)
-            : Contains(source, value, null);
+            source is ICollection<TSource> collection ? collection.Contains(value) :
+            Contains(source, value, null);
 
         public static bool Contains<TSource>(this IEnumerable<TSource> source, TSource value, IEqualityComparer<TSource> comparer)
         {
-            if (comparer == null)
-            {
-                comparer = EqualityComparer<TSource>.Default;
-            }
-
             if (source == null)
             {
                 throw Error.ArgumentNull(nameof(source));
             }
 
-            foreach (TSource element in source)
+            if (comparer == null)
```

Comment:
Would it be worth catching explicitly passed `EqualityComparer<TSource>.Default` too?

Reply:
Is that common?

Reply:
That's pretty much what I'm wondering. I would guess it would be more common with direct collection operations than with linq to always pass in a comparer and have `Default` as the default, but that's only a guess.

Reply:
I'm not going to right now. Other than size increase, this change doesn't penalize any cases (e.g. it's using a branch that's already there). Adding such a check in theory could negatively impact other cases, even if it might be negligible.